src/views/block_view.ts

```diff
@@ -48,84 +34,48 @@ export default class BlockView<BlockNode extends ManuscriptNode>
     this.updateContents()
   }
 
-  public updateContents = () => {
+  public updateContents() {
+    super.updateContents()
     this.updateClasses()
-    this.updateAttributes()
-    this.onUpdateContent()
+    this.updatePlaceholder()
   }
 
-  // unfortunately we can't call updateContents in successors because they are not methods but props
-  // which means they're inited in the constructor and are not accessible via super.
-  // onUpdateContent is provided here to allow to execute additional actions on content update without a need to copy all the code
-  // @TODO - rewrite arrow props to methods
-  onUpdateContent() {
-    return
+  handleTrackChanges() {
+    addTrackChangesAttributes(this.node.attrs, this.dom)
   }
 
-  public updateClasses = () => {
+  public updateClasses() {
     if (!this.contentDOM) {
       return
     }
-
     this.contentDOM.classList.toggle('empty-node', !this.node.childCount)
   }
 
-  public updateAttributes = () => {
+  public updatePlaceholder() {
     if (!this.contentDOM) {
       return
     }
-
-    const dataTracked = (this.node.attrs.dataTracked || []).filter(
-      (attr: TrackedAttrs) => attr.operation !== 'reference'
-    )
-
-    if (dataTracked?.length) {
-      const lastChange = dataTracked[dataTracked.length - 1]
-      this.dom.setAttribute('data-track-status', lastChange.status)
-      this.dom.setAttribute('data-track-op', lastChange.operation)
-    } else {
-      this.dom.removeAttribute('data-track-status')
-      this.dom.removeAttribute('data-track-op')
-    }
-
-    for (const [key, target] of Object.entries(this.viewAttributes)) {
-      if (key in this.node.attrs) {
-        const value = this.node.attrs[key]
-
-        if (value) {
-          this.contentDOM.setAttribute(target, value)
-        } else {
-          this.contentDOM.removeAttribute(target)
-        }
-      }
+    if (this.node.attrs.placeholder) {
+      this.contentDOM.dataset.placeholder = this.node.attrs.placeholder
     }
   }
 
-  public createElement = () => {
+  public createElement() {
     this.contentDOM = document.createElement(this.elementType)
     this.contentDOM.className = 'block'
-
     this.dom.appendChild(this.contentDOM)
   }
 
-  public createDOM = () => {
+  public createDOM() {
     this.dom = document.createElement('div')
     this.dom.classList.add('block-container')
     this.dom.classList.add(`block-${this.node.type.name}`)
   }
 
-  public createGutter = (className: string, buttons: HTMLElement[]) => {
+  public createGutter(className: string, buttons: HTMLElement[]) {
     const gutter = document.createElement('div')
     gutter.setAttribute('contenteditable', 'false')
     gutter.classList.add(className)
-    if (
-      isGraphicalAbstractFigure(
```

Comment:
@asouqi, note that I removed this, as it doesn't belong in this class. What's the purpose of it?

Reply:
this is to hide gutter buttons for GA figure. with your update we can do that in figure_element.ts`

Reply:
There is no need for an extra class for this. I updated the css.